diff --git a/src/filesystem_utils.h b/src/filesystem_utils.h
index b8da335..e761001 100644
--- a/src/filesystem_utils.h
+++ b/src/filesystem_utils.h
@@ -16,6 +16,10 @@
 #include <dirent.h>
 #endif // _WIN32

+#if __APPLE__
+#include <mach-o/dyld.h>
+#endif
+
 #if _WIN32
 typedef std::wstring path_t;
 #define PATHSTR(X) L##X
@@ -121,6 +125,18 @@ static path_t get_executable_directory()

     return path_t(filepath);
 }
+#elif __APPLE__
+static path_t get_executable_directory()
+{
+    char filepath[256];
+    uint32_t size = sizeof(filepath);
+    _NSGetExecutablePath(filepath, &size);
+
+    char* slash = strrchr(filepath, '/');
+    slash[1] = '\0';
+
+    return path_t(filepath);
+}
 #else // _WIN32
 static path_t get_executable_directory()
 {
